### PR TITLE
Handle errors at node-js

### PR DESCRIPTION
**Pull request by lamphamsy**
Error is handled in node-eclib.js instead of throwing it in cpp modules

### CODE REVIEW

File: node-eclib.js [hunk 1]
```diff
@@ -62,30 +62,38 @@ ECLib.prototype = {
...
+            err = {
+                errorcode: enums.ErrorCode.EINVALIDPARAMS,
+                message: util.getErrorMessage(enums.ErrorCode.EINVALIDPARAMS),
+            };
             instance_descriptor_id = err.errorcode;
         }
 
         if (!callback) {
+            if (instance_descriptor_id < 0) {
+                throw err.message;
```
Comment (ghost) [inline node-eclib.js:90]: Why throw a string here?

File: node-eclib.js [hunk 2]
```diff
@@ -108,9 +116,13 @@ ECLib.prototype = {
             err = util.getErrorMessage(enums.ErrorCode.EBACKENDNOTAVAIL);
         }
         if (!callback) {
+            if (err) {
+                throw err;
+                return;
+            }
             return err;
```
Comment (ghost) [inline node-eclib.js]: This always equates to `undefined`, doesn't it?

**Review by ghost — COMMENTED** [review]
I would like to know what motivated that change before committing to an answer. There is not enough context in this PR.

File: node-eclib.js (hunk 1 above)
Comment (lamphamsy) [reply node-eclib.js:90]: it shows error message that is useful for user, instead of all error code and message

**Comment by lamphamsy** [comment]
Merge to working branch

**Pull request merged**